src/test/java/org/jenkinsci/plugins/gitclient/GitClientTest.java

```diff
@@ -1441,6 +1442,7 @@ public void testSubmodulesUsedFromOtherBranches() throws Exception {
 
     @Issue("JENKINS-46054")
     @Test
+    @Ignore
```

Comment:
Is this one ignored because it can be considered tested by "SubmodulePatternStringTest"? Why not removing then?

Reply:
At least a comment pointing out the reason why the test is ignored (if the test is not removed as @varyvol suggests) would be useful.

Reply:
Thanks for detecting that mistake!

The test should *not* be ignored.  The submodule pattern string test is a nice unit test that confirms the regular expression based processing is working as expected.  This test is an integration test that performs an "end to end" verification of behavior.  During earlier attempts to fix the problem, I had cases where the unit tests of the pattern string were failing, and this integration test failed with a real problem that was not caught by the unit tests.

Reply:
Fixed by the next commit...